homeassistant/components/media_player/ziggo_mediabox_xl.py

```diff
@@ -0,0 +1,175 @@
+    """Set up the Ziggo Mediabox XL platform."""
+    from ziggo_mediabox_xl import ZiggoMediaboxXL
+
+    hass.data[DATA_KNOWN_DEVICES] = known_devices = set()
+
+    # Is this a manual configuration?
+    if config.get(CONF_HOST) is not None:
+        host = config.get(CONF_HOST)
+        name = config.get(CONF_NAME)
+    elif discovery_info is not None:
+        host = discovery_info.get('host')
+        name = discovery_info.get('name')
+    else:
+        _LOGGER.error("Cannot determine device")
+        return
+
+    # Only add a device once, so discovered devices do not override manual
+    # config.
+    hosts = []
+    ip_addr = socket.gethostbyname(host)
+    if ip_addr not in known_devices:
+        try:
+            mediabox = ZiggoMediaboxXL(ip_addr)
+            if mediabox.test_connection():
+                hosts.append(ZiggoMediaboxXLDevice(mediabox, host, name))
+                known_devices.add(ip_addr)
+            else:
+                _LOGGER.error("Can't connect to %s", host)
+        except socket.error as error:
+            _LOGGER.error("Can't connect to %s: %s", host, error)
+    else:
+        _LOGGER.info("Ignoring duplicate Ziggo Mediabox XL %s", host)
+    add_devices(hosts, True)
+
+
+class ZiggoMediaboxXLDevice(MediaPlayerDevice):
+    """Representation of a Ziggo Mediabox XL Device."""
+
+    def __init__(self, mediabox, host, name):
+        """Initialize the device."""
+        # Generate a configuration for the Samsung library
+        self._mediabox = mediabox
+        self._host = host
+        self._name = name
+        self._state = None
+
+    def update(self):
+        """Retrieve the state of the device."""
+        try:
+            if self._mediabox.turned_on():
+                if self._state != STATE_PAUSED:
+                    self._state = STATE_PLAYING
+            else:
+                self._state = STATE_OFF
+        except socket.error:
+            _LOGGER.error("Couldn't fetch state from %s", self._host)
+
+    def send_keys(self, keys):
+        """Send keys to the device and handle exceptions."""
+        try:
+            self._mediabox.send_keys(keys)
+        except socket.error:
+            _LOGGER.error("Couldn't send keys to %s", self._host)
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def source_list(self):
+        """List of available sources (channels)."""
+        return [self._mediabox.channels()[c]
+                for c in sorted(self._mediabox.channels().keys())]
+
+    @property
+    def supported_features(self):
+        """Flag media player features that are supported."""
+        return SUPPORT_ZIGGO
+
+    def turn_on(self):
+        """Turn the media player on."""
+        self.send_keys(['POWER'])
+        self._state = STATE_OFF
```

Comment:
Is this correct? Why set the state to `STATE_OFF` after turning the entity on?

Reply:
Oh wow, that makes little sense indeed. Swapped states to the correct ones.